foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiClient.java

```diff
@@ -95,6 +98,37 @@ public static OpenAiClient withCustomDestination(@Nonnull final Destination dest
     return new OpenAiClient(destination);
   }
 
+  /**
+   * Add a system prompt before user prompts.
+   *
+   * @param systemPrompt the system prompt
+   * @return the client
+   */
+  @Nonnull
+  public OpenAiClient withSystemPrompt(@Nonnull final String systemPrompt) {
+    parameters =
+        new OpenAiChatCompletionParameters()
+            .addMessages(new OpenAiChatSystemMessage().setContent(systemPrompt));
+    return this;
+  }
+
+  /**
+   * Generate a completion for the given user prompt.
+   *
+   * @param prompt a text message.
+   * @return the completion output
+   * @throws OpenAiClientException if the request fails
+   */
+  @Nonnull
+  public OpenAiChatCompletionOutput chatCompletion(@Nonnull final String prompt)
+      throws OpenAiClientException {
+    if (parameters == null) {
+      parameters = new OpenAiChatCompletionParameters();
+    }
+    parameters.addMessages(new OpenAiChatUserMessage().addText(prompt));
```

Comment:
This doesn't work, because the client may be used repeatedly for multiple chat completions. With this approach, old messages would accumulate.